lib/chef/provider/package/apt.rb

```diff
@@ -124,6 +124,11 @@ def unlock_package(name, version)
 
         private
 
+        # @return [String] package name with or without anchors attached to it.
+        def resolve_package(pkg)
+          new_resource.anchor_package_regex ? "^#{pkg}$" : pkg
+        end
+
```

Comment:
@thheinen  There were calls this action which need to be added back as well (Ref https://github.com/chef/chef/pull/14397/files#diff-a2ffb973f9e185cf5309ac78ed94c807edb322e81570fb40b3c55c70dccb5b0bL183)
Thanks.